testing-overhead/src/test/java/io/opentelemetry/agents/AgentResolver.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.agents;
+
+import java.io.FileOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.Optional;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class AgentResolver {
+
+  private final LatestAgentSnapshotResolver snapshotResolver = new LatestAgentSnapshotResolver();
+
+  public Optional<Path> resolve(Agent agent) throws Exception {
+    if(Agent.NONE.equals(agent)){
+      return Optional.empty();
+    }
+    if(Agent.LATEST_SNAPSHOT.equals(agent)){
```

Comment:
Instead of snapshot, which includes the annoying resolver, how about just building from this repo and using that agent, similar to our smoke tests? The snapshot can get old due to main build failures such as testLatestDeps so that would be easier to reason about which agent commit is actually teste

Reply:
I had really hoped to avoid this and to keep this test process decoupled from the main build.  It's going to slow things down considerably.

Is the thought that regressions would be harder to track back to commits or something?

Reply:
I support Jason in his reluctance to build local project every time.